### PR TITLE
try to make cleanup more robust

### PR DESCRIPTION
**Pull request by wouterdb**
# Description

Ignore cleanup failures in order to make tests more robust

Questions: does this require a changelog/version bump? (it doesn't change the actual code at all)

# Merge procedure

Don't use the github built-in merge, but the process described [here](https://docs.internal.inmanta.com/topics/tasks/commiting_changes_modules.html)

```sh
git pull
git checkout master
git pull
git merge --squash issue/{issue-number}-{short description}
inmanta module commit -m "{Commit Message Here}" -r
git push
git push {tag} # push the tag as well
```

Then close the PR with a reference to the commit

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] ~~Attached issue to pull request~~
- [ ] Changelog entry
- [ ] Version number is bumped to dev version
- [x] Code is clear and sufficiently documented
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [x] ~~End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )~~

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -199,6 +209,23 @@ def _cleanup(ec2, elb, resource_name_prefix: str):
...
+        (cleanup_vpc, (ec2, resource_name_prefix)),
+        (cleanup_ssh, (ec2, resource_name_prefix)),
+        (cleanup_elb, (elb, resource_name_prefix)),
+    ]
+    # ignore cleanup failures
+    for function, arguments in functions:
+        try:
+            function(*arguments)
+        except Exception:
+            LOGGER.exception("Failed to execute cleanup method %s", function.__name__)
```
Comment (arnaudsjs) [inline tests/conftest.py:226]: Shouldn't we print the exception for debugging purposes.

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: tests/conftest.py (hunk 1 above)
Comment (wouterdb) [reply tests/conftest.py:226]: it will (or should) (https://docs.python.org/3/library/logging.html#logging.Logger.exception)

Comment (arnaudsjs) [reply tests/conftest.py:226]: True. I missed the fact that it was logged as exception.

**Review by arnaudsjs — APPROVED** [review]

File: tests/conftest.py [hunk 2]
```diff
@@ -199,6 +209,23 @@ def _cleanup(ec2, elb, resource_name_prefix: str):
...
 
 
+def _cleanup(ec2, elb, resource_name_prefix: str):
+    functions = [
+        (cleanup_instances, (ec2, resource_name_prefix)),
+        (cleanup_volumes, (ec2, resource_name_prefix)),
+        (cleanup_igw, (ec2, resource_name_prefix)),
+        (cleanup_vpc, (ec2, resource_name_prefix)),
+        (cleanup_ssh, (ec2, resource_name_prefix)),
+        (cleanup_elb, (elb, resource_name_prefix)),
```
Comment (sanderr) [inline tests/conftest.py:219]: `functools.partial` would fit nicely here but I'm fine with this approach as well.

**Review by sanderr — APPROVED** [review]
I don't think this requires a version bump.

File: tests/conftest.py (hunk 2 above)
Comment (wouterdb) [reply tests/conftest.py:219]: I also needed the method name, so I reverted the partial approach.

**Pull request merged**